patchwork-events-rendering/src/main/java/net/patchworkmc/impl/event/render/RenderEvents.java

```diff
@@ -53,4 +60,23 @@ public static void onTextureStitchPre(SpriteAtlasTexture spriteAtlasTexture, Set
 	public static void onTextureStitchPost(SpriteAtlasTexture spriteAtlasTexture) {
 		eventDispatcher.accept(new TextureStitchEvent.Post(spriteAtlasTexture));
 	}
+
+	/**
+	 * Called by ForgeHooksClient and MixinWorldRenderer.
+	 * @return true if the bounding box rendering is cancelled.
+	 */
+	public static boolean onDrawHighlightEvent(WorldRenderer context, Camera info, HitResult target, int subID, float partialTicks) {
+		switch (target.getType()) {
+		case BLOCK:
+			if (!(target instanceof BlockHitResult)) return false;
+			return MinecraftForge.EVENT_BUS.post(new DrawBlockHighlightEvent.HighlightBlock(context, info, target, subID, partialTicks));
+		case ENTITY:
+			if (!(target instanceof EntityHitResult)) return false;
+			return MinecraftForge.EVENT_BUS.post(new DrawBlockHighlightEvent.HighlightEntity(context, info, target, subID, partialTicks));
+		default:
+			break;
+		}
+
+		return MinecraftForge.EVENT_BUS.post(new DrawBlockHighlightEvent(context, info, target, subID, partialTicks));
```

Comment:
This return could be moved into the default case instead of being outside the switch block.